### ISSUE TITLE
Documentation update

### ISSUE THREAD

**Issue by yx9o** [apache/shardingsphere#11387]
Hi, community, I found that some implementation classes of the ShardingSphereRuleBuilder (name has been modified) interface and YamlRuleConfigurationSwapper in the user development manual 5.2 configuration are not listed in the document, and the details are as follows:
```java
interface：RuleBuilder
Implementation classes：
AlgorithmProvidedDatabaseDiscoveryRuleBuilder
AlgorithmProvidedEncryptRuleBuilder
AlgorithmProvidedReadwriteSplittingRuleBuilder
AlgorithmProvidedShardingRuleBuilder
AuthorityRuleBuilder
DatabaseDiscoveryRuleBuilder
EncryptRuleBuilder
ReadwriteSplittingRuleBuilder
ShadowRuleBuilder
ShardingRuleBuilder
SingleTableRuleBuilder
```
```java
interface：YamlRuleConfigurationSwapper
Implementation classes：
AuthorityRuleConfigurationYamlSwapper
DatabaseDiscoveryRuleAlgorithmProviderConfigurationYamlSwapper
DatabaseDiscoveryRuleConfigurationYamlSwapper
EncryptRuleAlgorithmProviderConfigurationYamlSwapper
EncryptRuleConfigurationYamlSwapper
ReadwriteSplittingRuleAlgorithmProviderConfigurationYamlSwapper
ReadwriteSplittingRuleConfigurationYamlSwapper
ShadowRuleConfigurationYamlSwapper	
ShardingRuleAlgorithmProviderConfigurationYamlSwapper
ShardingRuleConfigurationYamlSwapper
```
problem:
1. Can the `ShardingSphereRuleBuilder` interface be adjusted to `RuleBuilder`?
2. Adjust `ReplicaQueryRuleBuilder` to `ReadwriteSplittingRuleBuilder`
3. Adjust `ReplicaQueryRuleConfigurationYamlSwapper` to `ReadwriteSplittingRuleConfigurationYamlSwapper`
4. Do implementation classes not listed in the document need to be supplemented?

If there is a need, I can do it.

**Comment by menghaoranss** [comment]
@yx9o  good suggestions， could you fix them with a PR

**Comment by yx9o** [comment]
ok

**Comment by menghaoranss** [comment]
fixed by #11399
